aptos-move/aptos-aggregator/src/types.rs

```diff
@@ -59,3 +61,98 @@ impl TryFrom<AggregatorVersionedID> for StateKey {
         }
     }
 }
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum AggregatorValue {
+    Aggregator(u128),
+    Snapshot(u128),
+    Derived(Vec<u8>),
```

Comment:
Why do we need derived? Having Snapshot(SnapshotValue) seems cleaner to me

Reply:
It's not clear to me, but worth exploring

I am not going to modify this at the bottom of the stack - but will try out at the top and we can discuss on the PR that shows how that would look like at the top.